src/main/java/com/google/api/codegen/discovery/MainDiscoveryProviderFactory.java

```diff
@@ -99,11 +98,14 @@ public static DiscoveryProvider defaultCreate(
           .setOutputRoot(outputRoot)
           .build();
     } else if (id.equals(NODEJS)) {
-      return CommonDiscoveryProvider.newBuilder()
-          .setContext(new NodeJSDiscoveryContext(service, apiaryConfig))
-          .setSnippetSetRunner(
-              new NodeJSSnippetSetRunner<Method>(SnippetSetRunner.SNIPPET_RESOURCE_ROOT))
-          .setSnippetFileName(id + "/" + DEFAULT_SNIPPET_FILE)
+      return ViewModelProvider.newBuilder()
```

Comment:
I think this is better than the original, but it still feels repetitive. You could have two hash map literals of the form
```
Map<String,Class> test = 
    new HashMap<String, Class>{
      NODEJS: NodeJSSampleMethodToViewTransformer.getClass(),
      ...}
```

and then have a single builder/return statement rather than the if chain (or switch. why wasn't the code using a switch to begin with?).

These hash maps could easily be static variables, BTW.

(And there might be ways to get away without even getClass, maybe?)

The advantage of this is that all the builder methods are called in one place, and the parameters that differ are very easy to visually inspect. And it's more concise.